aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -28,32 +28,51 @@
 
 #include <map>
 
+namespace at {
+namespace native {
+
 // NOTE:
-// YOU ARE NOT OBLIGED TO USE THESE MACROS
+// YOU ARE NOT OBLIGED TO USE THESE HELPERS
 // If you're writing something more specialized, please don't try to make them
 // work for your case, but just write something new instead.
 
-namespace at {
-namespace native {
+// Helper functions that reduce redundant code in implementing the most typical kind of unary operators. This allows
```

Comment:
nit: I'd reorder this a bit.  Basically you want to order of points to be:
1) these are for typical operators
2) therefore, don't add more specialized handling here, they are just for the common case
3) make the preprocessing point.